### ISSUE TITLE
Event Emitter for Angular animations 

### ISSUE THREAD

**Issue by priscilalima** [vitalii-andriiovskyi/ngx-owl-carousel-o#9]
Hi, I am trying to create an angular animation for a card component, but it is not working with the events emitted by the slider.

To create the angular state animations, I would need an event based on any change of the slider (dragging or clicking on the dots) and the state of the slide data (what slide is active).

I tried to use the current events, but when the user uses the dragging, but it doesn't complete the action it doesn't work. 

Maybe that's something that you can help me with.

Thanks.

**Comment by vitalii-andriiovskyi** [comment]
Hi.
I published `v1.0.3` for Angular 7 and `v0.1.2` for Angular 6. 
They add events `initialized` and `change`, also modify the payload of event `dragging`.
The previous `dragging` payload was `$event = true/false`. Now payload is: 
``` typescript
{
  dragging: boolean,
  data: SlidesOutputData
}

class SlidesOutputData {
  startPosition?: number;
  slides?: SlideModel[];
};
```

Events `initialized` and `change` emit `SlidesOutputData`. This object is defined before firing this event. In other words, it shows current active slides for the moment when user clicks dots or any of nav buttons. 

However, while the user drags the carousel this event fires after dropping the carousel or after stopping dragging.  For dragging you need to use the `dragging` event.

The order is: `dragging` (payload `{dragging: true, data: {...}`), -> `dragging` (payload: `{dragging: false, data: {...}`), -> `change` (payload: `SlidesOutputData`). All the time prop `data` and payload of  `change` are the same. To get less affected of this 3 following events,  you can use Observable (e.g. `BehaviorSubject`) and operator `distinctUntilChanged` which emits items from the source Observable with distinct values.

**Comment by priscilalima** [comment]
Thank you so much, i will update my code.
